### PR TITLE
Add MSSQL test cases

### PR DESCRIPTION
**Pull request by Biruntha**
## Automation tests
 - Unit tests 
   > Code coverage information
 - Integration tests
   > Details about the test cases and coverage

## Security checks
 - Followed secure coding standards in http://wso2.com/technical-reports/wso2-secure-engineering-guidelines? yes/no
 - Ran FindSecurityBugs plugin and verified report? yes/no
 - Confirmed that this PR doesn't commit any keys, passwords, tokens, usernames, or other secrets? yes/no

### CODE REVIEW

File: database/mssql/src/test/resources/sql-src/db-init.sql [hunk 1]
```diff
@@ -0,0 +1,7 @@
+IF NOT EXISTS (SELECT name FROM sys.databases WHERE name = N'simpledb')
+BEGIN
+    CREATE DATABASE simpledb;
```
Comment (Manuri) [inline database/mssql/src/test/resources/sql-src/db-init.sql]: Shall we use same db name as in MySQL and PostgreSQL - `testdb`?

File: database/mssql/src/test/java/org/ballerinalang/scenario/test/database/mssql/CallTest.java [hunk 2]
```diff
@@ -0,0 +1,250 @@
...
+    }
+
+    @Test(description = "Test cursor type")
+    public void testCallFunctionWithRefcursor() {
+        BValue[] returns = BRunUtil.invoke(callCompilerResult, "testCallFunctionWithRefcursor");
+        AssertionUtil.assertCallQueryReturnValue(returns[0]);
+        Assert.assertTrue(returns[1] instanceof BValueArray, "Invalid type");
+        BValueArray stringValues = (BValueArray) returns[1];
+        String[] fieldValues = {
+                "ABCD", "SQL Server VARCHAR", "ああ", "ありがとうございまし"
```
Comment (Manuri) [inline database/mssql/src/test/java/org/ballerinalang/scenario/test/database/mssql/CallTest.java]: Do we have a specific reason to insert japanese characters? Just wondering. :)

File: database/mssql/src/test/java/org/ballerinalang/scenario/test/database/mssql/SelectTest.java [hunk 3]
```diff
@@ -0,0 +1,227 @@
...
+        };
+        AssertionUtil.assertNonNullStringValues(dateTimeTypeRecord, 7, fieldValues, "id");
+    }
+
+    @Test(description = "Test nil date time type selection query", groups = { DISABLED })
+    public void testDateTimeTypesNil() {
+        BValue[] returns = BRunUtil.invoke(selectCompileResult, "testDateTimeTypesNil");
+        Assert.assertTrue(returns[0] instanceof BMap);
+        AssertionUtil.assertNullValues((BMap) returns[0], 7, "id");
+    }
```
Comment (Manuri) [inline database/mssql/src/test/java/org/ballerinalang/scenario/test/database/mssql/SelectTest.java]: Missing new line

File: database/mssql/src/test/resources/bal-src/call-test.bal [hunk 4]
```diff
@@ -0,0 +1,282 @@
...
+    if (ret is error?) {
+        return ret;
+    } else {
+        error e = error("Unexpected return type: table");
+        return e;
+    }
+}
+
+function testCallOutParamNumericTypes() returns @tainted [table<record{}>[]|error?, any[]] {
+
```
Comment (Manuri) [inline database/mssql/src/test/resources/bal-src/call-test.bal]: Additional new line.

**Comment by Manuri** [comment]
Let's change the UpdateTest to verify whether the data we inserted have been inserted as expected, in a separate PR, as discussed offline.

File: database/mssql/src/test/java/org/ballerinalang/scenario/test/database/mssql/CallTest.java [hunk 5]
```diff
@@ -0,0 +1,250 @@
...
+        callCompilerResult = BCompileUtil
+                .compile(Paths.get(resourcePath.toString(), "bal-src", "call-test.bal").toString());
+    }
+
+    @Test(description = "Test numeric type IN params")
+    public void testCallInParamNumericTypes() {
+        BValue[] returns = BRunUtil.invoke(callCompilerResult, "testCallInParamNumericTypes");
+        AssertionUtil.assertCallQueryReturnValue(returns[0]);
+    }
+
```
Comment (Manuri) [inline database/mssql/src/test/java/org/ballerinalang/scenario/test/database/mssql/CallTest.java]: Shall we separate numeric types in to following categories 
1. Integer types eg: INT, SMALLINT, BIGINT etc
2. Floating point types eg: FLOAT, DOUBLE etc
3. Fixed point types eg: DECIMAL, NUMERIC etc

Comment (Biruntha) [reply database/mssql/src/test/java/org/ballerinalang/scenario/test/database/mssql/CallTest.java]: Fixed in https://github.com/ballerina-platform/ballerina-scenario-tests/pull/38/commits/1402cee58bb3dd0339a86f09a6228ae0e8f64f1f

**Comment by Biruntha** [comment]
> Let's change the UpdateTest to verify whether the data we inserted have been inserted as expected, in a separate PR, as discussed offline.

fixed in https://github.com/ballerina-platform/ballerina-scenario-tests/pull/38/commits/1402cee58bb3dd0339a86f09a6228ae0e8f64f1f

**Pull request merged**